core/src/main/java/be/orbinson/aem/dictionarytranslator/services/DictionaryService.java

```diff
@@ -1,192 +1,131 @@
  * <p>
- * A high-level read-only API is provided by the {@link be.orbinson.aem.dictionarytranslator.models.Dictionary} model and by the {@link be.orbinson.aem.dictionarytranslator.services.impl.CombiningMessageEntryResourceProvider}.
+ * A high-level read-only API is provided  by the {@link be.orbinson.aem.dictionarytranslator.services.impl.CombiningMessageEntryResourceProvider}.
  *
  */
 @ProviderType
 public interface DictionaryService {
 
-    boolean isEditableDictionary(Resource dictionaryResource);
-    
-    /**
-     * Returns the ordinal of the dictionary resource. The lower the ordinal, the higher the precedence.
-     * The ordinal is determined by the path of the dictionary resource and follows the logic from
-     * <a href="https://sling.apache.org/documentation/bundles/internationalization-support-i18n.html#resourcebundle-hierarchies">ResourceBundle hierarchies</a>.
-     * The lowest ordinal is 0 and the highest depends on the number of search paths configured for the resource resolver.
-     * @param dictionaryResource the dictionary resource
-     * @return the ordinal of the dictionary resource
+    /** 
+     * Returns all dictionaries present in the system. Leverages a query to find all dictionaries.
+     * @param resourceResolver
+     * @return
      */
-    int getOrdinal(Resource dictionaryResource);
-
-    List<Resource> getDictionaries(ResourceResolver resourceResolver);
-
-    void createDictionary(Resource parent, String name, String[] languages, String basename) throws PersistenceException;
+    Collection<LanguageDictionary> getAllDictionaries(ResourceResolver resourceResolver);
 
-    void deleteDictionary(ResourceResolver resourceResolver, String dictionaryPath) throws DictionaryException, ReplicationException, PersistenceException;
-
-    /**
-     * Returns the list of languages for the given dictionary resource in alphabetical order.
-     * @param dictionaryResource
-     * @return the list of languages for the given dictionary resource in alphabetical order
+    /** 
+     * Returns all dictionaries present in the system in a map where the key is the parent path of the dictionary and the value is a sorted map of dictionaries by locale.
+     * Internally calls {@link #getAllDictionaries(ResourceResolver)}.
+     * @param resourceResolver
+     * @return
      */
-    List<String> getLanguages(Resource dictionaryResource);
-
-    void deleteLanguage(Resource dictionaryResource, String language) throws DictionaryException, ReplicationException, PersistenceException;
-
-    void addLanguage(Resource dictionaryResource, String language, String basename) throws PersistenceException;
-
-    String getBasename(Resource dictionaryResource);
-
-    List<String> getKeys(Resource dictionaryResource) throws DictionaryException;
+    Map<String, SortedMap<Locale, LanguageDictionary>> getAllDictionariesByContainer(ResourceResolver resourceResolver);
```

Comment:
is there a difference between container or parent path? In other docs and method names you use parent path instead of container

Reply:
No, there is no difference. I don't know yet what naming is better suited. Any preference?